src/_P155_CAN_helper.ino

```diff
@@ -0,0 +1,231 @@
+#include "_Plugin_Helper.h"
+#ifdef USES_P155
+
+// #######################################################################################################
+// ################################### Plugin 155: CAN SJA1000 Importer ##################################
+// #######################################################################################################
+
+
+# include "src/Helpers/_Plugin_Helper_serial.h"
+# include <CAN.h>
+
+
+# define PLUGIN_155
+# define PLUGIN_ID_155         155
+# define PLUGIN_NAME_155       "Communication - CAN Importer"
+
+# define P155_MAX_NODES      255
+# define P155_TOPIC_MAX_SIZE 16
+# define P155_TIMEOUT        50
+
+# define P155_VAL1_FLAG 0x01
+# define P155_VAL2_FLAG 0x02
+# define P155_VAL3_FLAG 0x04
+# define P155_VAL4_FLAG 0x10
+
+#define PLUGIN_VALUENAME1_155 "Status"
+
+void P155_check_timeout(EventStruct *event, int idx)
+{
+  int16_t idx_flag = (0x01 << idx);
+
+  if (PCONFIG_ULONG(idx) > PCONFIG(2))
+  {
+    if (!(PCONFIG(3) & idx_flag))
+    {
+      String log_debug = F("Disconnected");
+      addLogMove(LOG_LEVEL_INFO, log_debug);
+    }
+    PCONFIG(3) |= idx_flag;
+  }
+  else if (!(PCONFIG(3) & idx_flag))
+  {
+    PCONFIG_ULONG(idx) += 100;
+  }
+}
+
+boolean Plugin_155(uint8_t function, struct EventStruct *event, String& string)
+{
+  boolean success = false;
+
+  switch (function)
+  {
+    case PLUGIN_DEVICE_ADD:
+    {
+      Device[++deviceCount].Number           = PLUGIN_ID_155;
+      Device[deviceCount].Type               = DEVICE_TYPE_DUMMY;
+      Device[deviceCount].VType              = Sensor_VType::SENSOR_TYPE_NONE;
+      Device[deviceCount].Ports              = 0;
+      Device[deviceCount].PullUpOption       = false;
+      Device[deviceCount].InverseLogicOption = false;
+      Device[deviceCount].FormulaOption      = false;
+      Device[deviceCount].ValueCount         = 0;
+      Device[deviceCount].SendDataOption     = true;
+      Device[deviceCount].TimerOption        = false;
+      Device[deviceCount].GlobalSyncOption   = false;
+      Device[deviceCount].TimerOptional      = true;
+      Device[deviceCount].Custom             = true;
+      Device[deviceCount].DecimalsOnly       = true;
+      break;
+    }
+
+    case PLUGIN_GET_DEVICENAME:
+    {
+      string = F(PLUGIN_NAME_155);
+      break;
+    }
+
+    case PLUGIN_GET_DEVICEVALUENAMES:
+    {
+      const uint8_t valueCount = getValueCountFromSensorType(static_cast<Sensor_VType>(PCONFIG(1)));
+      if (valueCount > 0) {
+        String lines[valueCount];
+        LoadCustomTaskSettings(event->TaskIndex, lines, valueCount, 0);
+
+        for (int i = 0; i < valueCount; ++i) {
+        if (lines[i].length() == 0) {
+          lines[i] = concat(F("val"), i + 1);
+        }
+          ExtraTaskSettings.setTaskDeviceValueName(i, lines[i]);
+        }
```

Comment:
Please use Uncrustify to format the sources (exception: external libraries), to avoid mis-alignments like these.